src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/steps/prepare-step/prepare-step.component.ts

```diff
@@ -1,9 +1,117 @@
-import { ChangeDetectionStrategy, Component } from "@angular/core";
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Input,
+    OnInit,
+} from "@angular/core";
+import {
+    ControlContainer,
+    FormArray,
+    FormBuilder,
+    FormGroup,
+    FormGroupDirective,
+} from "@angular/forms";
+import { MaybeNull } from "src/app/common/app.types";
+import { BaseComponent } from "src/app/common/base.component";
+import { SetPollingSourceSection } from "src/app/shared/shared.types";
+import { EditFormType, PrepareKind } from "../../add-polling-source-form.types";
+import { EditPollingSourceService } from "../../edit-polling-source.service";
 
 @Component({
     selector: "app-prepare-step",
     templateUrl: "./prepare-step.component.html",
     styleUrls: ["./prepare-step.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
+    viewProviders: [
+        { provide: ControlContainer, useExisting: FormGroupDirective },
+    ],
 })
-export class PrepareStepComponent {}
+export class PrepareStepComponent extends BaseComponent implements OnInit {
+    @Input() public eventYamlByHash: MaybeNull<string> = null;
+    @Input() public sectionName: SetPollingSourceSection;
+    public parentForm: FormGroup;
+    public setPollingSourceEvent: MaybeNull<EditFormType> = null;
+
```

Comment:
I don't see any validation logic.
I've clicked 1 times add pipe, 1 times add decompress, and did not enter anything.
It's possible to preview empty incorrect steps in the YAML view:

```
  prepare:
    - kind: pipe
      command:
        - ""
    - kind: decompress
      format: ""
```

But final saving fails on the backend:
```
Error:
Serde error: unknown variant ``, expected `gzip` or `zip`
```

Reply:
Add validation